app/javascript/flavours/polyam/entrypoints/admin.tsx

```diff
@@ -285,6 +285,53 @@ async function mountReactComponent(element: Element) {
   );
 }
 
+// Polyam: System skin settings
+const onChangeFlavourAndSkin = (target: HTMLSelectElement) => {
+  const darkSelectElement = document.querySelector<HTMLSelectElement>(
+    'select#form_admin_settings_system_dark',
+  );
+
+  const lightSelectElement = document.querySelector<HTMLSelectElement>(
+    'select#form_admin_settings_system_light',
+  );
+
+  const selectedGroup = target.selectedOptions[0]
+    .parentNode as HTMLOptGroupElement;
+
+  if (darkSelectElement) {
+    for (const child of darkSelectElement.children) {
+      if (child instanceof HTMLOptGroupElement) {
+        if (child.label !== selectedGroup.label) {
+          child.disabled = true;
+        } else {
+          child.disabled = false;
+        }
+      }
+    }
+  }
+
+  if (lightSelectElement) {
+    for (const child of lightSelectElement.children) {
+      if (child instanceof HTMLOptGroupElement) {
+        if (child.label !== selectedGroup.label) {
+          child.disabled = true;
+        } else {
+          child.disabled = false;
+        }
+      }
+    }
+  }
```

Comment:
I don't like the repeating code, but it is in line with "keeping it simple".
It just feels wrong to do.